app.js

```diff
@@ -282,7 +282,9 @@
       },
 
       getStringQuery: function(type) {
-        return new RegExp(this.$('.query.' + type).val(), 'i');
+        var query = this.$('.query.' + type).val();
+        var escapedQuery = query.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
```

Comment:
If it works it's good for me!

On Tue, Nov 3, 2015 at 5:33 PM, Joe Palmieri <notifications@github.com>
wrote:

> In app.js
> <https://github.com/jpalmieri/best_search/pull/13#discussion_r43832654>:
>
> > @@ -282,7 +282,9 @@
> >        },
> >
> >        getStringQuery: function(type) {
> > -        return new RegExp(this.$('.query.' + type).val(), 'i');
> > +        var query = this.$('.query.' + type).val();
> > +        var escapedQuery = query.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
>
> Although I didn't plan it this way, I think so; I think it won't match if
> it doesn't escape the space too.
> https://gyazo.com/c0683cb6a92f4d253640f1434ceefeb7
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jpalmieri/best_search/pull/13/files#r43832654>.
>